sql/core/src/test/scala/org/apache/spark/sql/ExpressionsSchemaSuite.scala

```diff
@@ -0,0 +1,202 @@
+        sys.env.contains("SPARK_HOME"), "spark.test.home or SPARK_HOME is not set.")
+      sys.props.getOrElse("spark.test.home", sys.env("SPARK_HOME"))
+    }
+
+    java.nio.file.Paths.get(sparkHome,
+      "sql", "core", "src", "test", "resources", "sql-functions").toFile
+  }
+
+  private val resultFile = new File(baseResourcePath, "sql-expression-schema.md")
+
+  val ignoreSet = Set(
+    // Random output without a seed
+    "org.apache.spark.sql.catalyst.expressions.Rand",
+    "org.apache.spark.sql.catalyst.expressions.Randn")
+
+  /** A single SQL query's SQL and schema. */
+  protected case class QueryOutput(
+    className: String,
+    funcName: String,
+    sql: String = "N/A",
+    schema: String = "N/A") {
+    override def toString: String = {
+      s"| $className | $funcName | $sql | $schema |"
+    }
+  }
+
+  test("Check schemas for expression examples") {
+    val exampleRe = """^(.+);\n(?s)(.+)$""".r
+    val funInfos = spark.sessionState.functionRegistry.listFunction().map { funcId =>
+      spark.sessionState.catalog.lookupFunctionInfo(funcId)
+    }
+
+    val classFunsMap = funInfos.groupBy(_.getClassName).toSeq.sortBy(_._1)
+    val outputBuffer = new ArrayBuffer[String]
+    val outputs = new ArrayBuffer[QueryOutput]
+    val missingExamples = new ArrayBuffer[String]
+
+    classFunsMap.foreach { kv =>
+      val className = kv._1
+      if (!ignoreSet.contains(className)) {
+        kv._2.foreach { funInfo =>
+          val example = funInfo.getExamples
+          val funcName = funInfo.getName.replaceAll("\\|", "&#124;")
+          if (example == "") {
+            val queryOutput = QueryOutput(className, funcName)
+            outputBuffer += queryOutput.toString
+            outputs += queryOutput
+            missingExamples += funcName
+          }
+
+          // If expression exists 'Examples' segment, the first element is 'Examples'. Because
+          // this test case is only used to print aliases of expressions for double checking.
+          // Therefore, we only need to output the first SQL and its corresponding schema.
+          // Note: We need to filter out the commands that set the parameters, such as:
+          // SET spark.sql.parser.escapedStringLiterals=true
+          example.split("  > ").tail.filterNot(_.trim.startsWith("SET")).take(1).foreach {
+            case exampleRe(sql, _) =>
+              val df = spark.sql(sql)
+              val escapedSql = sql.replaceAll("\\|", "&#124;")
+              val schema = df.schema.catalogString.replaceAll("\\|", "&#124;")
+              val queryOutput = QueryOutput(className, funcName, escapedSql, schema)
+              outputBuffer += queryOutput.toString
+              outputs += queryOutput
+            case _ =>
+          }
+        }
+      }
+    }
+
+    if (regenerateGoldenFiles) {
+      val missingExampleStr = missingExamples.mkString(",")
+      val goldenOutput = {
+        "Automatically generated by ExpressionsSchemaSuite\n" +
```

Comment:
nit: `s"<!-- Automatically generated by${getClass.getSimpleName} -->\n"`

Reply:
You mean just as comment not to display?

Reply:
It's OK.